src/Cake.Common/Tools/NUnit/NUnit3Runner.cs

```diff
@@ -0,0 +1,219 @@
+                throw new ArgumentNullException("settings");
+            }
+
+            Run(settings, GetArguments(assemblyPaths, settings));
+        }
+
+        private ProcessArgumentBuilder GetArguments(IEnumerable<FilePath> assemblyPaths, NUnit3Settings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            // Add the assemblies to build.
+            foreach (var assemblyPath in assemblyPaths)
+            {
+                builder.AppendQuoted(assemblyPath.MakeAbsolute(_environment).FullPath);
+            }
+
+            if (settings.Test != null)
+            {
+                builder.Append("--test=" + settings.Test);
+            }
+
+            if (settings.TestList != null)
+            {
+                builder.AppendQuoted(string.Format(CultureInfo.InvariantCulture, "--testlist={0}", settings.TestList.MakeAbsolute(_environment).FullPath));
+            }
+
+            if (settings.Where != null)
+            {
+                builder.Append("--where \"" + settings.Where + "\"");
+            }
+
+            if (settings.Timeout.HasValue)
+            {
+                builder.Append("--timeout=" + settings.Timeout.Value);
+            }
+
+            if (settings.Seed.HasValue)
+            {
+                builder.Append("--seed=" + settings.Seed.Value);
+            }
+
+            if (settings.Workers.HasValue)
+            {
+                builder.Append("--workers=" + settings.Workers.Value);
+            }
+
+            if (settings.StopOnError)
+            {
+                builder.Append("--stoponerror");
+            }
+
+            if (settings.Work != null)
+            {
+                builder.AppendQuoted(string.Format(CultureInfo.InvariantCulture, "--work={0}", settings.Work.MakeAbsolute(_environment).FullPath));
+            }
+
+            if (settings.OutputFile != null)
+            {
+                builder.AppendQuoted(string.Format(CultureInfo.InvariantCulture, "--out={0}", settings.OutputFile.MakeAbsolute(_environment).FullPath));
+            }
+
+            if (settings.ErrorOutputFile != null)
+            {
+                builder.AppendQuoted(string.Format(CultureInfo.InvariantCulture, "--err={0}", settings.ErrorOutputFile.MakeAbsolute(_environment).FullPath));
+            }
+
+            if (settings.Full)
+            {
+                builder.Append("--full");
+            }
+
+            if (settings.ResultsFile != null && settings.NoResults)
+            {
+                throw new ArgumentException(
+                    GetToolName() + ": You can't specify both a results file and set NoResults to true.");
+            }
+
+            if (settings.ResultsFile != null)
+            {
+                builder.AppendQuoted(
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        "--result={0}", settings.ResultsFile.MakeAbsolute(_environment).FullPath));
+            }
```

Comment:
This needs to be updated. I forgot that we also support `--result=filename.xml;format=nunit2` and other modifiers. I will push a change today.